nesta/core/luigihacks/misctools.py

```diff
@@ -125,3 +125,18 @@ def extract_task_info(luigi_task):
     task_name = type(luigi_task).__name__
     routine_id = f'{task_name}-{luigi_task.date}-{test}'
     return test, routine_id
+
+
+@lru_cache()
+def bucket_keys(bucket_name):
```

Comment:
Note, this function is introduced to replace boilerplate in a bunch of luigi routines where batchables are marked as `done` if a given key exists (i.e. S3 is used as an intermediate checkpoint).